### PR TITLE
ensure ingress-nginx node label is applied

### PR DESCRIPTION
**Pull request by nabuskey**
fixes: #366 
fixes: #367

### CODE REVIEW

File: pkg/kind/cluster.go [hunk 1]
```diff
@@ -241,28 +246,61 @@ func (c *Cluster) ensureCorrectConfig(in []byte) (kindv1alpha4.Cluster, error) {
...
+					}
+				}
+				break nodes
+			}
+		}
+		if node.Labels != nil {
+			v, ok := node.Labels[ingressNginxNodeLabelKey]
+			if ok && v == ingressNginxNodeLabelValue {
+				appendIngresNodeLabel = false
+				nodePosition = i
```
Comment (nimakaviani) [inline pkg/kind/cluster.go:279]: I wonder if this can be further simplified. I am not fond of label breaks in go, but leave it up to you.

**Review by nimakaviani — CHANGES_REQUESTED** [review]
just a minor typo and a comment

File: pkg/kind/cluster.go [hunk 2]
```diff
@@ -241,28 +246,61 @@ func (c *Cluster) ensureCorrectConfig(in []byte) (kindv1alpha4.Cluster, error) {
 	if err != nil {
 		return kindv1alpha4.Cluster{}, fmt.Errorf("parsing kind config: %w", err)
 	}
-
+	// the port and ingress-nginx label must be on the same node to ensure nginx runs on the node with the right port.
 	appendNecessaryPort := true
+	appendIngresNodeLabel := true
```
[the remark this answers is not in the document]
Comment (nabuskey) [reply pkg/kind/cluster.go]: fixed.

File: pkg/kind/cluster.go (hunk 1 above)
Comment (nabuskey) [reply pkg/kind/cluster.go]: It's really just comes down to last one winning or the first one winning. I chose first one winning. If we do last one wins, we can avoid using label breaks.

Comment (nimakaviani) [reply pkg/kind/cluster.go]: i like first one winning too.

Here is an alternative way of writing it (thanks ChatGPT). I dont think it is any easier tho. let's leave it.

```
for i, node := range parsedCluster.Nodes {
    checkNodeLabels := func() {
        if node.Labels != nil {
            if v, ok := node.Labels[ingressNginxNodeLabelKey]; ok && v == ingressNginxNodeLabelValue {
                appendIngresNodeLabel = false
            }
        }
    }

    for _, pm := range node.ExtraPortMappings {
        if strconv.Itoa(int(pm.HostPort)) == c.cfg.Port {
            appendNecessaryPort = false
            nodePosition = i
            checkNodeLabels()
            break
        }
    }

    if appendNecessaryPort { // If the port was not matched, check labels
        checkNodeLabels()
        if !appendIngresNodeLabel {
            nodePosition = i
            break
        }
    }
}
```

**Review by nimakaviani — APPROVED** [review]
LGTM

**Pull request merged**